src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php

```diff
@@ -85,15 +85,20 @@ protected function removePrices(ProductInterface $product, AttributeInterface $a
     {
         $productValue = $product->getValue($attribute->getCode(), $locale, $scope);
 
+        $currencyToRemove = [];
+        foreach ($data as $priceToRemove) {
+            $currencyToRemove[] = $priceToRemove['currency'];
+        }
+
         if (null !== $productValue) {
             $prices = [];
             foreach ($productValue->getData() as $price) {
-                if (!in_array(['amount' => $price->getData(), 'currency' => $price->getCurrency()], $data)) {
```

Comment:
👀  In my fix, I only check for the currency to remove the price value. not the price *and* the value.

What do you think ?